rust/src/lib/ifaces/inter_ifaces_controller.rs

```diff
@@ -157,6 +157,59 @@ impl MergedInterfaces {
         Ok(())
     }
 
+    // If port name reference should be check kernel name first, then fallback
+    // to profile name.
+    pub(crate) fn resolve_port_name_ref(&mut self) -> Result<(), NmstateError> {
+        let mut profile_2_kernel: HashMap<String, String> = HashMap::new();
+        for iface in self.kernel_ifaces.values() {
+            let base_iface = iface.merged.base_iface();
+            if let Some(profile_name) = base_iface.profile_name.as_deref() {
+                if profile_name != base_iface.name.as_str() {
+                    profile_2_kernel.insert(
+                        profile_name.to_string(),
+                        base_iface.name.to_string(),
+                    );
+                }
+            }
+        }
+        for iface in self.iter_mut() {
+            let des_iface = if let Some(d) = iface.desired.as_mut() {
+                d
+            } else {
+                continue;
+            };
+            if !des_iface.is_up() {
+                continue;
+            }
+            let ports: Vec<String> = if let Some(ports) = des_iface.ports() {
+                ports.iter().map(|p| p.to_string()).collect()
```

Comment:
You don't need the `to_string` here. Let `ports` to be a `Vec<&str>`.

Reply:
```rust
                    des_iface
                        .change_port_name(&port_name, kernel_name.to_string());
```

The `des_iface` will be changed later, so you cannot hold reference to it.